eng/scripts/Language-Settings.ps1

```diff
@@ -5,31 +5,117 @@ $packagePattern = "*.zip"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/python-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=python%2F&delimiter=%2F"
 
-function Get-python-PackageInfoFromRepo  ($pkgPath, $serviceDirectory, $pkgName)
-{
-  pip install packaging==20.4 -q -I
-  $pkgName = $pkgName.Replace('_', '-')
-  if (Test-Path (Join-Path $pkgPath "setup.py"))
+function Get-python-PackageInfoFromRepo  ($pkgPath, $serviceDirectoryName, $artifactName)
+{  
+  $packageName = $artifactName.Replace('_', '-')
+  $pkgDirName = Split-Path $pkgPath -Leaf
+  if ($pkgDirName -ne $packageName)
+  {
+    # Common code triggers this function against each directory but we can skip if it doesn't match package name
+    return $null
+  }
+
+  $setupPyPath = Join-Path $pkgPath "setup.py"
+
+  if (Test-Path $setupPyPath)
   {
-    $setupLocation = $pkgPath.Replace('\','/')
-    pushd $RepoRoot
-    $setupProps = (python -c "import sys; import os; sys.path.append(os.path.join('scripts', 'devops_tasks')); from common_tasks import get_package_properties; obj=get_package_properties('$setupLocation'); print('{0},{1},{2}'.format(obj[0], obj[1], obj[2]));") -split ","
-    popd
-    if (($setupProps -ne $null) -and ($setupProps[0] -eq $pkgName))
+    # Get PackageName and Version Info from SetUp.py
+    $packageNameFromSetupPy = $null
+    $namespaceNameFromSetupPy = $null
+    $versionFromSetupPy = $null
+    $versionFileSubDir = $null
+    $requiresCorePkg = $False
+    $nameAndVersionInfo = Select-String -Path $setupPyPath -Pattern '^\s+version=.+,$', '^\s+name=.+,$', '^PACKAGE_NAME\s=.+$', '^NAMESPACE_NAME\s=.+$', '^package_folder_path\s=.+$'-Raw
```

Comment:
Is it worth filtering these here given you are matching on them below? It is probably better to avoid the duplicate regexs.